gateway/src/service.rs

```diff
@@ -549,6 +552,35 @@ impl GatewayService {
         }
     }
 
+    pub async fn check_project_count(
+        &self,
+        account_name: &AccountName,
+        is_admin: bool,
+        is_pro: bool,
+    ) -> Result<(), Error> {
+        if is_admin {
+            return Ok(());
+        }
+
+        let proj_count: u32 = query("SELECT COUNT(*) FROM projects WHERE account_name = ?1")
```

Comment:
Nit: would it be more efficient to count the primary key?
```suggestion
        let proj_count: u32 = query("SELECT COUNT(project_name) FROM projects WHERE account_name = ?1")
```

Reply:
Shouldn't matter. It just counts the matching rows AFAIK.